### ISSUE TITLE
Protected Entities

### ISSUE THREAD

**Issue by Cooldude2606** [explosivegaming/scenario#209]
**Feature overview**
Name: Protected entities
TL;DR: Certain entities, such as water pumps, become protected when placed and can only be removed by the owner and regulars. 

**Details**
Within the config file there will be a set of entities that once placed will become protected and can not be removed by guests unless they palced it. This can include water pumps, steam engines, train stops etc. If a player attempts to mine one of these entities then a warning is shown in chat for online mods to see. Mods will also have a command which can be used to set a single entity as protected, and again to remove the protection.

**Possible methods**
Have a global array of protected entitites that is searched when a player with guest rank mines an entitiy. If they are not the owner then the entity is replaced and a message is displayed to mods.

**Additional context**
Maybe it would be better to have the warnings as alerts next to the toolbar. It should also be considered if the entity should be replaced by the script, as doing so would make greifers aware of the system and find another less detectable way to greif.

**Comment by Cooldude2606** [comment]
Feature has been implemented in #213 and merged into dev branch. Closing issue.
